feature/experimental/p4rt/ate_tests/google_discovery_protocol_packetout_test/google_discovery_protocol_packetout_test.go

```diff
@@ -0,0 +1,326 @@
+	component := telemetry.Component{}
+	component.IntegratedCircuit = &telemetry.Component_IntegratedCircuit{}
+	component.Name = ygot.String(*p4rtNodeName)
+	component.IntegratedCircuit.NodeId = ygot.Uint64(deviceId)
+	dut.Config().Component(*p4rtNodeName).Replace(t, &component)
+}
+
+// configurePortId configures p4rt port-id on the DUT.
+func configurePortId(ctx context.Context, t *testing.T, dut *ondatra.DUTDevice) {
+	ports := sortPorts(dut.Ports())
+	for i, port := range ports {
+		dut.Config().Interface(port.Name()).Id().Replace(t, uint32(i)+portId)
+	}
+}
+
+// setupP4RTClient sends client arbitration message for both leader and follower clients,
+// then sends setforwordingpipelineconfig with leader client.
+func setupP4RTClient(ctx context.Context, args *testArgs) error {
+	// Setup p4rt-client stream parameters
+	streamParameter := p4rt_client.P4RTStreamParameters{
+		Name:        streamName,
+		DeviceId:    deviceId,
+		ElectionIdH: uint64(0),
+		ElectionIdL: electionId,
+	}
+
+	// Send ClientArbitration message on both p4rt leader and follower clients.
+	clients := []*p4rt_client.P4RTClient{args.leader, args.follower}
+	for index, client := range clients {
+		if client != nil {
+			client.StreamChannelCreate(&streamParameter)
+			if err := client.StreamChannelSendMsg(&streamName, &p4_v1.StreamMessageRequest{
+				Update: &p4_v1.StreamMessageRequest_Arbitration{
+					Arbitration: &p4_v1.MasterArbitrationUpdate{
+						DeviceId: streamParameter.DeviceId,
+						ElectionId: &p4_v1.Uint128{
+							High: streamParameter.ElectionIdH,
+							Low:  streamParameter.ElectionIdL - uint64(index),
+						},
+					},
+				},
+			}); err != nil {
+				return errors.New("Errors seen when sending ClientArbitration message.")
+			}
+			if _, _, arbErr := client.StreamChannelGetArbitrationResp(&streamName, 1); arbErr != nil {
+				return errors.New("Errors seen in ClientArbitration response.")
+			}
+		}
+	}
+
+	// Load p4info file.
+	p4Info, err := utils.P4InfoLoad(p4InfoFile)
+	if err != nil {
+		return errors.New("Errors seen when loading p4info file.")
+	}
+
+	// Send SetForwardingPipelineConfig for p4rt leader client.
+	if err := args.leader.SetForwardingPipelineConfig(&p4_v1.SetForwardingPipelineConfigRequest{
+		DeviceId:   deviceId,
+		ElectionId: &p4_v1.Uint128{High: uint64(0), Low: electionId},
+		Action:     p4_v1.SetForwardingPipelineConfigRequest_VERIFY_AND_COMMIT,
+		Config: &p4_v1.ForwardingPipelineConfig{
+			P4Info: &p4Info,
+			Cookie: &p4_v1.ForwardingPipelineConfig_Cookie{
+				Cookie: 159,
+			},
+		},
+	}); err != nil {
+		return errors.New("Errors seen when sending SetForwardingPipelineConfig.")
+	}
+	return nil
+}
+
+// getGDPParameter returns GDP related parameters for testPacketOut testcase.
+func getGDPParameter(t *testing.T) PacketIO {
+	return &GDPPacketIO{
+		IngressPort: fmt.Sprint(portId),
+	}
+}
+
+func TestPacketOut(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	ctx := context.Background()
+
+	// Configure the DUT
```

Comment:
Remove the redundant comment as configureDUT(t, dut) is very clear.

Reply:
sure. changes are done and pushed.